parser/prism/Factory.h

```diff
@@ -0,0 +1,109 @@
+#ifndef SORBET_PARSER_PRISM_FACTORY_H
+#define SORBET_PARSER_PRISM_FACTORY_H
+
+#include "common/common.h"
+#include "core/LocOffsets.h"
+#include <string_view>
+#include <vector>
+extern "C" {
+#include "prism.h"
+}
+
+namespace sorbet::parser::Prism {
+
+// Forward declarations
+class Parser;
+
+class Factory {
+private:
+    Parser &parser;
+
+public:
+    Factory(Parser &parser) : parser(parser) {}
+
+    template <typename T> T *allocateNode() const {
+        return new T{};
+    }
```

Comment:
This should be allocating things with Prism's allocator, not the general C++ one.